TASK-51 Completed: Creation of a new page called leaderboard

Asafyaco opened the pull request:
Created a new page called **leader-board.page** - display all the top 10 users with the most score on the game

* Created a **new route of the LeaderBoardPage** on App.jsx
* Updated the translation both on English & Hebrew - on navBar and the title of the page
* Updated the translation.service
* Added to the navBarComponent in the floating button the page - **LeaderBoard** 
* Integrate with an API on the backend
* Added on states on constants.ts - **/leaderBoard**

Example of the page -
![image](https://user-images.githubusercontent.com/93646445/167867361-b256974b-3b07-47a0-a9c9-2110d3ad9ad5.png)

amitimo committed:
TASK-51 Completed: Creation of a new page called leaderboard

Asafyaco commented:
* Added an emoji next to the title of the page on **leaderBoardPage**
* Added media query to navBar component that **wrap all the links** and don't hide them depending on the device
* Updated dependencies on the **leaderBoardPage.scss** 

Example of warped navBar on different devices:
![image](https://user-images.githubusercontent.com/93646445/167867593-55925485-d0a1-4d12-a5a9-6ace43b2e4c9.png)

Example of floating button on navbar:
![image](https://user-images.githubusercontent.com/93646445/167867683-97900232-9fe6-4f07-9f52-740665204e06.png)

FadiFarah commented on src/pages/leader-board/leader-board.scss:
```diff
@@ -0,0 +1,61 @@
+    text-transform: uppercase;
+    overflow-y: scroll;
+    line-height: normal;
+    height: 80vh;
```
change the height to height: calc(100vh - 10em); as we did on the other pages for the scroll to work properly.

Asafyaco replied:
👍

FadiFarah commented on src/pages/leader-board/leader-board.page.jsx:
```diff
@@ -0,0 +1,40 @@
+    <div className="wrapper">
+      <div className="board-wrapper qcg-flex qcg-flex-column qcg-flex-align-center">
+        <h1>{translationService.translate.leaderBoardPage.title}</h1>
+        {leaderboardUsers.length <= 10 ? (leaderboardUsers.map((top10, index)=>
```
change the naming of top10 to user.

FadiFarah commented on src/pages/leader-board/leader-board.page.jsx:
```diff
@@ -0,0 +1,40 @@
+    <div className="wrapper">
+      <div className="board-wrapper qcg-flex qcg-flex-column qcg-flex-align-center">
+        <h1>{translationService.translate.leaderBoardPage.title}</h1>
+        {leaderboardUsers.length <= 10 ? (leaderboardUsers.map((top10, index)=>
```
We don't need an if statement. We always get from the database maximum 10 users, so need to check. Remove it.

Asafyaco replied:
Ohh ok, didn't remember that. Has been removed

FadiFarah commented on src/pages/leader-board/leader-board.page.jsx:
```diff
@@ -0,0 +1,40 @@
+          <div className="points"> {top10.score}</div>
+          </div>})):(
+             <div></div>
+           )} 
```
Remove the unneeded empty div with the above multiple opening and closing brackets.

FadiFarah commented on src/assets/translations/english.json:
```diff
@@ -8,6 +8,7 @@
       "about": "about",
       "rules": "rules",
       "profile": "profile",
+      "leaderBoard":"leader board",
```
remove the space between the two words only for English.

FadiFarah commented on src/pages/leader-board/leader-board.scss:
```diff
@@ -0,0 +1,61 @@
+
+    @media only screen and (max-width: 940px) and (max-height: 660px) {
+      &.qcg-leader-board-page {
+        height: 55vh;
```
No need for height here, remove it.

FadiFarah reviewed (changes requested):
One additional thing to change... the leaderboard should only be visible to authenticated users only, just as the profile.

Asafyaco commented:
**Commit updated by comments**

* Changed height on leader-boards.scss
* Updated name on map function on leader-board.page
* Removed an if statement & Removed empty div
* Removed height on media query 
* Updated that the leader board page will shown only to the authenticated users
* Updated translation of first letter of the links on floating button on navBarComponent to a capital letter

FadiFarah committed:
TASK-51 completed: additional UI fixes
